internal/devpkg/package.go

```diff
@@ -121,7 +121,7 @@ func PackageFromStringWithOptions(raw string, locker lock.Locker, opts devopt.Ad
 	pkg := PackageFromStringWithDefaults(raw, locker)
 	pkg.DisablePlugin = opts.DisablePlugin
 	pkg.PatchGlibc = opts.PatchGlibc
-	pkg.outputs = initOutputs(opts.Outputs)
+	pkg.initOutputs(opts.Outputs)
```

Comment:
line 117 above needs `pkg.initOutputs`, for when the `raw` is a flake-ref-with-flake-outputs